components/style/values/specified/length.rs

```diff
@@ -1009,3 +1009,13 @@ impl Parse for LengthOrPercentageOrAutoOrContent {
 }
 
 pub type LengthOrNumber = Either<Length, Number>;
+
+impl LengthOrNumber {
+    pub fn parse_non_negative_number(input: &mut Parser) -> Result<Self, ()> {
+        if let Ok(v) = input.try(|i| Length::parse(i)) {
```

Comment:
The [original implementation](https://github.com/servo/servo/pull/14261/files#diff-c18e92fdea89758412a30a0e3f4e0213L1038) did a plain parse for Length and a parse_non_negative only for Number.  Was that incorrect, too?